clients/venice-admin-tool/src/main/java/com/linkedin/venice/ZkCopier.java

```diff
@@ -110,6 +159,54 @@ static TreeNode buildRequiredPathsTree(Set<String> clusterNames, String basePath
     return root;
   }
 
+  /**
+   * Build a sorted list of Venice-specific paths in the ZK client filtered by 1)base path, 2)cluster names, and 3)required cluster ZK paths.
+   * @return a sorted list of Venice-specific paths
+   */
+  static List<String> getZkClientPathsList(ZkClient zkClient, Set<String> clusterNames, String basePath) {
+    List<String> paths = new ArrayList<>();
+    StringBuilder sb = new StringBuilder(basePath);
+    paths.add(sb.toString());
+    String storeConfigsPath = sb.append("/").append(STORE_CONFIGS).toString();
+    if (zkClient.exists(storeConfigsPath)) {
+      paths.add(storeConfigsPath);
+      getZkClientPathsListHelper(zkClient, paths, sb);
+    }
+    sb.delete(sb.lastIndexOf("/" + STORE_CONFIGS), sb.length());
+    for (String cluster: clusterNames) {
+      String clusterPath = sb.append("/").append(cluster).toString();
+      if (zkClient.exists(clusterPath)) {
+        paths.add(clusterPath);
+        for (String venicePath: CLUSTER_ZK_PATHS) {
+          String clusterVenicePath = sb.append("/").append(venicePath).toString();
+          if (zkClient.exists(clusterVenicePath)) {
+            paths.add(clusterVenicePath);
+            getZkClientPathsListHelper(zkClient, paths, sb);
+          }
+          sb.delete(sb.lastIndexOf("/" + venicePath), sb.length());
+        }
+      }
+      sb.delete(sb.lastIndexOf("/" + cluster), sb.length());
+    }
+    Collections.sort(paths);
+    return paths;
+  }
+
+  /**
+   * Recursively traverse the ZK client and add each path to the list.
+   * @Note: This method directly modifies the list {@code paths} in the parent method {@code getZkClientPathsList()}
+   */
+  private static void getZkClientPathsListHelper(ZkClient zkClient, List<String> paths, StringBuilder sb) {
```

Comment:
I suppose the comment in the previous function about not using the same `StringBuilder` also transitively change this method's implementation. Still calling it out as it is not easy to make sense of this method just by taking a quick look at it.

Also, let's rename `sb` to something more representative

Reply:
Previous iteration was better https://github.com/linkedin/venice/pull/1056#discussion_r1683600582